### ISSUE TITLE
eager_gan.R

### ISSUE THREAD

**Issue by slfan2013** [rstudio/keras#1018]
The following error occurred and nothing happens.

WARNING:tensorflow:Gradients do not exist for variables ['batch_normalization/moving_mean:0', 'batch_normalization/moving_variance:0', 'batch_normalization_1/moving_mean:0', 'batch_normalization_1/moving_variance:0', 'batch_normalization_2/moving_mean:0', 'batch_normalization_2/moving_variance:0'] when minimizing the loss.

**Comment by henry090** [comment]
Please, apply the following changes. It has to work https://github.com/rstudio/keras/pull/1019/commits/39c2f04b69068513acccd83000329d9a4f0db19c

**Comment by skeydan** [comment]
Just re-added  2 lines that got lost in the process of the update to TF 2.
However, that does not result in a big speedup.

That example, as well as the other ones starting with `eager_`, were written more than 1.5 years ago, when eager execution was brand new, and not at all the normal way to do things, as it is since TF 2. Put differently, the blog posts written at that time were "pioneering examples" showing the new technique. 

Some of those have been ported to TF 2, but  - due to time constraints - just in a sense of "technically they run with TF 2". Performance-wise, there will most always be room for improvements! In fact, if someone wanted to contribute, the documentation on the website shows a bunch of great new techniques, especially this page:

https://tensorflow.rstudio.com/tutorials/advanced/

PRs improving the examples with these techniques are very welcome!

Closing this; again thanks @henry090 for your PR!
